src/pudl/analysis/record_linkage/classify_plants_ferc1.py

```diff
@@ -0,0 +1,309 @@
+        fuel_fractions[["utility_id_ferc1", "plant_name_ferc1", "report_year"] + ffc],
+        on=["utility_id_ferc1", "plant_name_ferc1", "report_year"],
+        how="left",
+    )
+
+    fuel_cols = list(ferc1_steam_df.filter(regex=".*_fraction_mmbtu$").columns)
+
+    # Train the classifier using DEFAULT weights, parameters not listed here.
+    clf = CrossYearLinker(
+        **{
+            "id_column": "plant_id_ferc1",
+            "column_transforms": [
+                ColumnTransform(
+                    **{
+                        "step_name": "plant_name_ferc1",
+                        "columns": "plant_name_ferc1",
+                        "transformer": "string",
+                        "weight": 2.0,
+                    }
+                ),
+                ColumnTransform(
+                    **{
+                        "step_name": "plant_type",
+                        "columns": ["plant_type"],
+                        "transformer": "category",
+                        "weight": 2.0,
+                        "cleaning_ops": ["null_to_empty_str"],
+                    }
+                ),
+                ColumnTransform(
+                    **{
+                        "step_name": "construction_type",
+                        "columns": ["construction_type"],
+                        "transformer": "category",
+                        "cleaning_ops": ["null_to_empty_str"],
+                    }
+                ),
+                ColumnTransform(
+                    **{
+                        "step_name": "capacity_mw",
+                        "columns": ["capacity_mw"],
+                        "transformer": "number",
+                        "cleaning_ops": ["null_to_zero"],
+                    }
+                ),
+                ColumnTransform(
+                    **{
+                        "step_name": "construction_year",
+                        "columns": ["construction_year"],
+                        "transformer": "category",
+                        "cleaning_ops": ["fix_int_na"],
+                    }
+                ),
+                ColumnTransform(
+                    **{
+                        "step_name": "utility_id_ferc1",
+                        "columns": ["utility_id_ferc1"],
+                        "transformer": "category",
+                    }
+                ),
+                ColumnTransform(
+                    **{
+                        "step_name": "fuel_fraction_mmbtu",
+                        "columns": fuel_cols,
+                        "transformer": Pipeline(
+                            [("scaler", MinMaxScaler()), ("norm", Normalizer())]
+                        ),
+                        "cleaning_ops": ["null_to_zero"],
+                    }
+                ),
+            ],
+        }
+    )
+    ferc1_steam_df = clf.fit_predict(ferc1_steam_df)
+
+    # Set the construction year back to numeric because it is.
```

Comment:
Should the NA values that were filled with blank strings or 0's be reverted or is that no longer necessary? In the old module, that happens in `revert_filled_in_string_nulls` and `revert_filled_in_float_nulls` of the old `classify_plants_ferc1` module

Reply:
Good call